### PR TITLE
Add RFDiffusionModel TorchModel wrapper

### PR DESCRIPTION
**Pull request by chidwipak**
## Description

Adds `RFDiffusionModel`, a `TorchModel` wrapper that lets you train and sample
from the RFDiffusion protein backbone model using standard DeepChem workflows.

Key things it adds:

- `model.fit(dataset)` — overrides `default_generator` to implement the DDPM
  training loop: normalize coords, pad to batch length, sample random timestep,
  add noise with `CosineSchedule.q_sample`, and yield
  `[noisy_coords, timestep, mask], [noise], [weights]`.
- `model.generate(num_samples, seq_length)` — runs the full reverse diffusion
  loop from Gaussian noise to backbone coordinates, with optional
  denormalization using statistics accumulated during training.
- `save_checkpoint` / `restore` — saves and restores the normalization stats
  alongside the model weights so generate() stays consistent after reloading.
- Optional `self_conditioning` — the model can condition on its own previous x0
  prediction during both training and sampling, following the RFDiffusion paper.

**Depends on PR #4996** (backbone denoiser) → #4995 (noise schedule) → #4994
(embeddings). This is PR D, the final PR in the 4-part split.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/rfdiffusion.py [hunk 1]
```diff
@@ -1,429 +1,407 @@
...
+            time_dim=time_dim,
+            num_layers=num_layers,
+            num_heads=num_heads,
+            max_seq_len=max_seq_len,
+            dropout=dropout,
+        )
+        # Attach schedule so it moves with model.to(device)
+        model.schedule = self.schedule
+
+        def diffusion_loss(outputs: List, labels: List,
```
Comment (JoseAntonioSiguenza) [inline deepchem/models/torch_models/rfdiffusion.py]: Please add complete docstrings

**Comment by JoseAntonioSiguenza** [comment]
LGTM

**Review by rbharath — APPROVED** [review]
LGTM
